batchlib/analysis/merge_tables.py

```diff
@@ -0,0 +1,113 @@
+import numpy as np
+
+from ..base import BatchJobOnContainer
+from ..util import get_logger, open_file
+
+
+logger = get_logger('Workflow.BatchJob.MergeAnalysisTables')
+
+
+class MergeAnalysisTables(BatchJobOnContainer):
+    """ Merge analysis tables written by CellLevelAnalysis for different parameters.
+    """
+    image_table_name = 'images/default'
+    well_table_name = 'wells/default'
+
+    def __init__(self, input_table_names, reference_table_name, **super_kwargs):
+
+        if reference_table_name not in input_table_names:
+            raise ValueError(f"{reference_table_name} was not found in {input_table_names}")
+
+        self.input_table_names = input_table_names
+        self.reference_table_name = reference_table_name
+
+        in_table_keys = ['images/' + in_key for in_key in input_table_names]
+        in_table_keys += ['wells/' + in_key for in_key in input_table_names]
+
+        # we store the global tables with .hdf5 ending to keep them separate from image files
+        in_pattern = '*.hdf5'
+        super().__init__(input_pattern=in_pattern,
+                         input_key=in_table_keys,
+                         input_format=['table'] * len(in_table_keys),
+                         output_key=[self.image_table_name,
+                                     self.well_table_name],
+                         output_format=['table', 'table'],
+                         **super_kwargs)
+
+    # we only keep columns that match the following names:
+    # q0.5* for all tables, also outlier and number of cells for the reference table
+    # see also https://github.com/hci-unihd/batchlib/issues/91
+    def _get_column_mask(self, column_names, is_reference_table):
+        # TODO should be parameters with reasonable defaults
+        # do we want the robost_z_score ?
+        # common_name_patterns = ('q0.5', 'robust_z_score')
+        common_name_patterns = ('q0.5',)
+        reference_name_patterns = ('score', 'n_cells', 'n_infected', 'n_control', 'fraction_infected',
+                                   'is_outlier', 'outlier_type')
+
+        keep_ids = [ii for ii, name in enumerate(column_names)
+                    if any(pattern in name for pattern in common_name_patterns)]
+        if is_reference_table:
+            keep_ids.extend([ii for ii, name in enumerate(column_names)
+                             if any(pattern in name for pattern in reference_name_patterns)])
+
+        col_mask = np.zeros(len(column_names), dtype='bool')
+        col_mask[keep_ids] = 1
+        return col_mask
+
+    def _format_col_name(self, name, prefix):
+        # TODO should we do something more fancy, like removing 'serum'?
```

Comment:
Yeah, why not remove 'serum' with `prefix=prefix.replace('serum','')`

Reply:
Yes, I think we should do this. I wanted to wait for @tischi to take a quick look at the initial version and see if there's other name changes we should do.